src/v2/Components/Modal/Modal.tsx

```diff
@@ -48,6 +50,9 @@ export class Modal extends React.Component<ModalProps> {
 
   close = () => {
     this.props.onClose()
+    if (this.props.goBackOnClose) {
```

Comment:
Just noticed the line above. Rather than adding a new prop, why not use `props.onClose` to handle backward and navigation, etc, can all be handled in the parent, keeping this compont dumb?

Reply:
I second this one, we should wrap into the parent components `onClose` function.

Reply:
Thanks for the advice and for pairing on this. I've pushed up the changes for this in f41beeb. I still need to complete the test for it...